server/api-manager.js

```diff
@@ -926,11 +926,35 @@ class MindbodyQueries {
     let funcc = () => {
       return request.makeRequest();
     };
-    const backoff = (retries, fn, delay = 500) => fn().catch(err => (retries > 1 && !this.atLimit()) ? pause(delay).then(() => backoff(retries - 1, fn, delay * 2)) : Promise.reject(err));
+    function pause(milliseconds) {
+      return new Promise(resolve => {setTimeout(resolve, milliseconds)});
+    };
+    const maxRetries = 5;
+    // Madison's original definition of backoff:
+    //const backoff = (retries, fn, delay = 500) => fn().catch(err => (retries > 1 && !this.atLimit()) ? pause(delay).then(() => backoff(retries - 1, fn, delay * 2)) : Promise.reject(err));
+    const backoff = (retries, fn, delay = 500) => {
```

Comment:
I "exploded" Madison's single line thing into this abominable long thing so that I could insert console logs during debugging. It _should_ do the same as the compact thing, but maybe I missed something in the translation.